app/src/main/java/com/example/marvel/di/ComicsModule.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.marvel.di
+
+import com.example.marvel.presentation.epoxy.ComicsController
+import com.example.marvel.presentation.ComicsViewModel
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.core.context.loadKoinModules
+import org.koin.dsl.module
+
+object ComicsModule {
+
+    private val comicsModule = module {
+        viewModel { ComicsViewModel() }
+        single { ComicsController() }
```

Comment:
Um `single` significa que por baixo dos panos, só vai haver uma instância do `ComicsController`. Ele é um [singleton](https://pt.wikipedia.org/wiki/Singleton) de luxo, enquanto a implementação do singleton padrão não é thread-safe, por baixo dos panos ele é implementado de uma forma `thread-safe`.

Não diria que esse é o comportamento esperado nesse caso. O controller guarda uma referência pro RecyclerView, e o RecyclerView guarda uma referência pro controller. Essa dependência cíclica, se não tratada, causa um [memory leak](https://stackoverflow.com/questions/35520946/leak-canary-recyclerview-leaking-madapter).

Quando a View morrer e for recriada, o `ComicsController` deveria ser recriado junto com ela.

Se for necessário guardar o estado da lista, podemos utilizar o ViewModel, ou o `onSaveInstanceState`.

Também acho DI incrível pra [Services, mas não é necessário pra um controller, na minha opinião.](https://github.com/mqueiroz1995/notebooks/blob/master/Computer%20Science/DependencyInjection.md#longer-introduction)